src/projectscene/view/trackspanel/trackitem.cpp

```diff
@@ -230,12 +240,20 @@ void TrackItem::setBalance(int balance)
     emit outputParamsChanged(m_outParams);
 }
 
+void TrackItem::commitBalance()
+{
+    projectHistory()->pushHistoryState(muse::TranslatableString("playback", "Moved pan dial").translated().toStdString(),
+                                       muse::TranslatableString("playback",
+                                                                "Pan").translated().toStdString(), trackedit::UndoPushType::CONSOLIDATE);
+}
+
 void TrackItem::setSolo(bool solo)
 {
     if (m_outParams.solo == solo) {
         return;
     }
 
+    trackPlaybackControl()->setSolo(trackId(), solo);
     m_outParams.solo = solo;
 
     // project::IProjectSoloMuteState::SoloMuteState soloMuteState;
```

Comment:
if we have implemented solo/mute state, then I suggest deleting all commented code about solo/mute